### PR TITLE
common: Using stl min | max, MIN | MAX macros instead

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
separate commit for reviewer(s).

File: src/common/bit_vector.hpp [hunk 1]
```diff
@@ -309,7 +309,7 @@ void BitVector<_b>::encode_data(bufferlist& bl, uint64_t byte_offset,
 
   uint64_t end_offset = byte_offset + byte_length;
   while (byte_offset < end_offset) {
-    uint64_t len = MIN(BLOCK_SIZE, end_offset - byte_offset);
+    uint64_t len = std::min((uint64_t)BLOCK_SIZE, end_offset - byte_offset);
```
Comment (amitkumar50) [inline src/common/bit_vector.hpp]: @shinobu-x This PR covers replacement of std::min by MIN.
https://github.com/ceph/ceph/pull/18124
Why we are moving them back?

Comment (liewegas) [reply src/common/bit_vector.hpp]: both are moving MIN -> std::min right?

File: src/common/buffer.cc [hunk 2]
```diff
@@ -1369,7 +1369,7 @@ using namespace ceph;
   uint32_t buffer::list::iterator_impl<is_const>::crc32c(
     size_t length, uint32_t crc)
   {
-    length = MIN( length, get_remaining());
+    length = std::min((size_t)length, (size_t)get_remaining());
```
Comment (cbodley) [inline src/common/buffer.cc]: tip: you can avoid casts by specifying which template specialization of std::min() you want:
```c++
length = std::min<size_t>(length, get_remaining());
```

**Comment by jdurgin** [comment]
closing since the explicit types have likely changed in bufferlist, and there are conflicts

**Pull request closed without merging**